contracts/orderbook/Logic/OrderBook.sol

```diff
@@ -0,0 +1,425 @@
+            srcToken = order.base;
+        } else {
+            srcToken = order.loanTokenAddress;
+        }
+        if (_orderExpiration[orderID] < block.timestamp) {
+            return false;
+        }
+        if (order.orderType == IOrderBook.OrderType.LIMIT_OPEN) {
+            if (order.loanID != 0 && !isActiveLoan(order.loanID)) {
+                return false;
+            }
+            if (srcToken == order.loanTokenAddress) {
+                amountUsed = (order.loanTokenAmount * order.leverage) / 10**18; //adjusts leverage
+            } else {
+                amountUsed = queryRateReturn(
+                    order.base,
+                    order.loanTokenAddress,
+                    order.collateralTokenAmount
+                );
+                amountUsed = (amountUsed * order.leverage) / 10**18;
+            }
+            (uint256 dSwapValue, ) = swapImpl.dexAmountOutFormatted(
+                payload,
+                amountUsed
+            );
+
+            if (order.amountReceived <= dSwapValue && dSwapValue > 0) {
+                return true;
+            }
+        } else if (order.orderType == IOrderBook.OrderType.LIMIT_CLOSE) {
+            if (!isActiveLoan(order.loanID)) {
+                return false;
+            }
+            uint256 dSwapValue;
+            if (order.isCollateral) {
+                (dSwapValue, ) = swapImpl.dexAmountOutFormatted(
+                    payload,
+                    order.collateralTokenAmount
+                );
+            } else {
+                (dSwapValue, ) = swapImpl.dexAmountInFormatted(
+                    payload,
+                    order.collateralTokenAmount
+                );
+            }
+            if (order.amountReceived <= dSwapValue) {
+                return true;
+            }
+        } else {
+            if (!isActiveLoan(order.loanID)) {
+                return false;
+            }
+            bool operand;
+            if (_useOracle[trader]) {
+                operand =
+                    order.amountReceived >=
+                    queryRateReturn(
+                        order.base,
+                        order.loanTokenAddress,
+                        order.collateralTokenAmount
+                    ); //TODO: Adjust for precision
+            } else {
+                operand =
+                    order.amountReceived >=
+                    getDexRate(
+                        swapImpl,
+                        order.base,
+                        order.loanTokenAddress,
+                        payload,
+                        order.collateralTokenAmount
+                    );
+            }
+            if (operand) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function getDexRate(
+        ISwapsImpl swapImpl,
+        address base,
+        address loanTokenAddress,
+        bytes memory payload,
+        uint256 amountIn
+    ) public returns (uint256 rate) {
```

Comment:
view?

Reply:
cant be as dexAmountOutFormatted is not view.